src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -1359,6 +1372,10 @@ def update_vm(cmd, resource_group_name, vm_name, os_disk=None, disk_caching=None
     if license_type is not None:
         vm.license_type = license_type
 
+    if user_data is not None:
+        from azure.cli.core.util import b64encode
+        vm.user_data = b64encode(user_data)
```

Comment:
Why not `read_content_if_is_file(user_data)` here?

Reply:
Before calling `build_vm_resource()` to build ARM template, method `read_content_if_is_file(user_data)` has been called to get the content of file or strings. [code link](https://github.com/Azure/azure-cli/blob/aa9ac1b2272a8ebd23272075f9bd0068030bf2df/src/azure-cli/azure/cli/command_modules/vm/custom.py#L928
) So here is the content of `user_data` that needs to be sent to REST.
This approach is consistent with `custom_data`.